test/modules/aws-cloudwatch-integration.ts

```diff
@@ -34,51 +44,124 @@ describe('AwsCloudwatch Integration Testing', () => {
+    VALUES (
+        '${logGroupName}', (SELECT region FROM aws_regions WHERE is_default = false and is_enabled = true LIMIT 1)
+    );
+  `));
+
+  it('applies creation of two log groups with the same name, but different regions', apply());
 
-  it('check no new log group', query(`
+  it('ARNs and regions for the two log groups with the same name should be different', query(`
     SELECT *
-    FROM log_group
-    WHERE log_group_name = '${logGroupName}';
-  `, (res: any[]) => expect(res.length).toBe(0)));
-  
-  it('adds a new log group', query(`
-    INSERT INTO log_group (log_group_name)
-    VALUES ('${logGroupName}');
+    FROM log_group;
+  `, (res: any) => {
+    // two log groups
+    expect(res.length).toBe(2);
+    // have non-empty ARNs (came from the cloud)
+    expect(res[0].log_group_arn !== '');
+    expect(res[1].log_group_arn !== '');
+    // their ARNs are not equal
+    expect(res[0].log_group_arn !== res[1].log_group_arn).toBe(true);
+    // but they have the same name (AWS does not allow duplicate name for log groups in a region)
+    expect(res[0].log_group_name === res[1].log_group_name).toBe(true);
+  }));
+
+  it('deletes the log group from the non-default region', query(`
+    DELETE FROM log_group
+    WHERE log_group_name = '${logGroupName}' AND region != default_aws_region();
   `));
 
-  it('check adds a new log group', query(`
-    SELECT *
-    FROM log_group
-    WHERE log_group_name = '${logGroupName}';
-  `, (res: any[]) => expect(res.length).toBe(1)));
+  it('syncs the state with the cloud to make sure it gets the resource from non-default region', sync());
 
-  it('applies the log group change', apply());
+  it('checks if the log group from the non-default region is back', query(`
+    SELECT * FROM log_group
+    WHERE log_group_name = '${logGroupName}' AND region != default_aws_region();
+  `, (res: any) => expect(res.length).toBe(1)));
 
-  it('uninstalls the cloudwatch module', uninstall(modules));
+  it('deletes the log group from the non-default region, this time for real', query(`
+    DELETE FROM log_group
+    WHERE log_group_name = '${logGroupName}' AND region != default_aws_region();
+  `));
 
-  it('installs the cloudwatch module', install(modules));
+  it('applies the deletion of the log group from the non-default region', apply());
 
-  it('tries to update a log group autogenerated field', query(`
-    UPDATE log_group SET log_group_arn = '${logGroupName}2' WHERE log_group_name = '${logGroupName}';
-  `));
+  it('syncs the state with the cloud', sync());
 
-  it('applies the log group change which will undo the change', apply());
+  it('checks if the log group in the default region is still there', query(`
+    SELECT * FROM log_group
+    WHERE log_group_name = '${logGroupName}' AND region = default_aws_region();
+  `, (res: any) => expect(res.length).toBe(1)));
 
-  it('deletes the log group', query(`
+  it('checks if the log group in the non-default region is gone', query(`
```

Comment:
good tests, I would add another test changing the log group from one region to another.